recipes-kernel/linux/files/patches-6.1/0100-swiotlb-Make-IO_TLB_SEGSIZE-configurable.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Li Hua Qian <huaqian.li@siemens.com>
+Date: Tue, 3 Dec 2024 14:49:59 +0800
+Subject: [PATCH] swiotlb: Make IO_TLB_SEGSIZE configurable
+
+In some applications, the default value of 128 is not sufficient for
+memory allocation and can cause runtime errors. This change makes
+IO_TLB_SEGSIZE configurable, allowing it to be increased if needed.
+
+Signed-off-by: Li Hua Qian <huaqian.li@siemens.com>
+---
+ include/linux/swiotlb.h | 6 +++++-
+ kernel/dma/Kconfig      | 7 +++++++
+ 2 files changed, 12 insertions(+), 1 deletion(-)
+
+diff --git a/include/linux/swiotlb.h b/include/linux/swiotlb.h
+index 35bc4e281c21..bf1e1f5cd67b 100644
+--- a/include/linux/swiotlb.h
++++ b/include/linux/swiotlb.h
+@@ -22,7 +22,11 @@ struct scatterlist;
+  * must be a power of 2.  What is the appropriate value ?
+  * The complexity of {map,unmap}_single is linearly dependent on this value.
+  */
+-#define IO_TLB_SEGSIZE	128
++#ifdef CONFIG_SWIOTLB_SEGSIZE
```

Comment:
Unneeded, the default is always defined, isn't it?

Reply:
not always, if we don't include the secureboot build, it won't be defined.

Reply:
Right, I set the default value as 128, then unneeded.